### PR TITLE
Add announcement banner

### PR DESCRIPTION
**Pull request by mashehu**
Screenshot 2024-01-22 at 16 59 06

Looks at `announcement_*` frontmatter in events or blog posts and displays the one with the start date closest to now.

### CODE REVIEW

File: src/content/config.ts [hunk 1]
```diff
@@ -58,12 +87,25 @@ const blog = defineCollection({
             label: z.array(z.string()),
             pubDate: z.date(),
             authors: z.array(z.string()),
+            announcement_start: z.date().optional(),
```
Comment (ewels) [inline src/content/config.ts:90]: No validation? 😉

File: src/content/events/2023/hackathon-october-2023.mdx [hunk 2]
```diff
@@ -2,7 +2,7 @@
 title: Hackathon - October 2023 (Barcelona)
 subtitle: A hybrid hackathon held in Barcelona and online
 type: hackathon
-start_announcement: '2023-07-13T12:00+02:00'
+start_announcement: 2023-07-13T12:00:00+02:00
```
[the remark this answers is not in the document]
Comment (mashehu) [reply src/content/events/2023/hackathon-october-2023.mdx:5]: It makes more sense for me to have "announcement" as a prefix for the different frontmatter entries.

File: src/content/config.ts (hunk 1 above)
Comment (mashehu) [reply src/content/config.ts:90]: uncommented 🙃

File: src/content/events/2023/hackathon-october-2023.mdx [hunk 3]
```diff
@@ -2,7 +2,8 @@
 title: Hackathon - October 2023 (Barcelona)
 subtitle: A hybrid hackathon held in Barcelona and online
 type: hackathon
-start_announcement: '2023-07-13T12:00+02:00'
+announcment:
```
Comment (ewels) [inline src/content/events/2023/hackathon-october-2023.mdx:5]: @mashehu - typo here

File: src/content/events/2023/hackathon-october-2023.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ewels) [reply src/content/events/2023/hackathon-october-2023.mdx]: Yup 👍🏻  My comment was that it looked like this one file hadn't been updated in line with the others. Sorry, should have given more context than just `?`.. 🙄

File: src/content/events/2023/hackathon-october-2023.mdx (hunk 3 above)
Comment (ewels) [reply src/content/events/2023/hackathon-october-2023.mdx:5]: Any idea why this didn't throw an error? The collections config should have been upset, no?

Comment (mashehu) [reply src/content/events/2023/hackathon-october-2023.mdx:5]: It's not a required parameter and it doesn't care about addition frontmatter data.

Comment (ewels) [reply src/content/events/2023/hackathon-october-2023.mdx:5]: :shakefist:

Comment (mashehu) [reply src/content/events/2023/hackathon-october-2023.mdx:5]: fixed in #2226

**Pull request merged**